spec.md

```diff
@@ -0,0 +1,442 @@
+platform/vendor specific protocols (AWS Kinesis, Azure Event Grid).
+
+## Context Attributes
+Every event in the OpenEventing specification includes context which is a set
+of consistent metadata attributes tools and developers can rely upon to
+understand how to handle the event and its data.
+
+Context is designed to be delivered separately from the event data (e.g. in
+protocol headers or protocol specific attributes).  This allows the context
+to be analyzed without having to deserialize the event data.  The context can
+also be serialized with the event data (e.g. a JSON implementation may
+consist of one JSON object that contains both context and data).
+
+### namespace
+* Type: String
+* Description: Identifier that uniquely identifies the organization publishing
+  the event.
+* Examples:
+  * kafka.apache.org
+  * com.microsoft.azure
+
+### event-type
+* Type: String
+* Description: Type of the event.  Producers can specify the format of this,
+  depending on their service.  This enables the interpretation of data, and
+  can be used for routing, policy and more.
+* Examples:
+  * customer.created
+* Constraints:
+  * Required
+* Notes:
+  * It is up for discussion whether namespace could be included as a prefix in
+    event-type.
+  * It is up for discussion whether the event-type should also include a
+    version.  It’s currently included in a separate attribute titled
+    event-type-version.
+  * It is up for discussion whether this specification should enforce a type
+    format.
+
+### event-type-version
+* Type: String
+* Description: The version of the event-type.  This enables the interpretation
+  of data by eventual consumers, requires the consumer to be knowledgeable about
+  the producer.
+* Notes:
+  * It is up for discussion whether this is applicable to the whole event or
+    to the data payload alone.  Currently, event-type-version covers the data,
+    while open-events-version covers the context.
+
+### open-events-version
+* Type: String
+* Description: The version of the OpenEventing specification which the event
+  uses.  This enables the interpretation of the context.
+
+### resource
+* Type: Object
+* Description: This describes the software instance that emits the event at
+  runtime (i.e. the producer).  It contains sub-properties (listed below)
+
+
+### resource-type
+* Type: String
+* Description: Type of the event source. Providers define list of event sources.
+* Constraints:
+  * Required
+* Examples:
+  * s3
+
+### resource-id
+* Type: String
+* Description: ID of the event source.
+* Constraints:
+  * Required
+*Examples:
+  *my.s3.bucket
+
+### event-id
+* Type: String
+* Description: ID of the event.  Can be specified by the producer.  The
+  semantics of this string are explicitly undefined to ease the implementation
+  of producers.  Enables deduplication.
+* Examples:
+  * A database commit ID
+* Constraints:
+  * Required
+  * Unique per producer
+
+### correlation-id
```

Comment:
@duglin Could we move this to the `Context Attributes Backlog`.  This was suggested in the Google doc by a contributor, but wasn't agreed to by anyone else.  We need to discuss and review new properties before adding them in.